core/vm/contracts.go

```diff
@@ -1048,3 +1062,65 @@ func (c *bls12381MapG2) Run(input []byte) ([]byte, error) {
 	// Encode the G2 point to 256 bytes
 	return g.EncodePoint(r), nil
 }
+
+type remoteStaticCall struct{}
+
+func (c *remoteStaticCall) RequiredGas(input []byte) uint64 {
+	return params.RemoteStaticCallGas
+}
+
+func (c *remoteStaticCall) Run(ctx PrecompileContext, input []byte) ([]byte, error) {
+	rpcUrl := ctx.GetL1ArchiveRpc()
+	if rpcUrl == nil {
+		return nil, errors.New("L1 archive RPC is not configured")
+	}
+
+	rpcCtx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
```

Comment:
This is a bit risky, if a single request times out then you'll end up on a fork. We probably don't want a timeout since we trust the remote node anyways, preferring safety over liveness

Reply:
Ok sounds good, will just remove timeout